packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -332,13 +332,122 @@ export class MeshDebugPluginMaterial extends MaterialPluginBase {
     @serialize()
     private _isEnabled: boolean;
 
-    private _options: Required<MeshDebugOptions>;
+    private _mode: MeshDebugMode;
     /**
-     * Options for the plugin.
-     * See MeshDebugOptions interface for defaults.
+     * The mesh debug visualization.
+     * Defaults to NONE.
+     */
+    @serialize()
+    @expandToProperty("_markAllDefinesAsDirty")
+    public mode: MeshDebugMode;
+
+    private _multiply: boolean;
+    /**
+     * Whether the mesh debug visualization should multiply with color underneath.
+     * Defaults to true.
+     */
+    @serialize()
+    @expandToProperty("_markAllDefinesAsDirty")
+    public multiply: boolean;
+
+    private _shadedDiffuseColor: Color3;

```

Comment:
In fact, you don't need to rebuild the shader when this property changes, so you don't need the `@expandToProperty("_markAllDefinesAsDirty")` construction.

You can simply replace:
```typescript
    private _shadedDiffuseColor: Color3;
    @serializeAsColor3()
    @expandToProperty("_markAllDefinesAsDirty")
    public shadedDiffuseColor: Color3;
```
with:
```typescript
    @serializeAsColor3()
    public shadedDiffuseColor: Color3;
```
Same for most properties.

I think you only need the `mode` and `multiply` update to actually recompile the shader.

Reply:
Fixed :)